plugins/woocommerce-admin/client/products/fills/more-menu-items/delete-variation-menu-item.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * External dependencies
+ */
+import { useParams } from 'react-router-dom';
+import { MenuGroup, MenuItem } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { useMemo, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME,
+	ProductAttribute,
+	ProductVariation,
+} from '@woocommerce/data';
+import { getNewPath, navigateTo } from '@woocommerce/navigation';
+import { RemoveConfirmationModal } from '@woocommerce/product-editor';
+import { recordEvent } from '@woocommerce/tracks';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { useEntityId, useEntityProp } from '@wordpress/core-data';
+
+export type DeleteVariationMenuItemProps = { onClose(): void };
+
+export const DeleteVariationMenuItem = ( {
+	onClose,
+}: DeleteVariationMenuItemProps ) => {
+	const [ showModal, setShowModal ] = useState( false );
+
+	const { productId } = useParams();
+
+	const variationId = useEntityId( 'postType', 'product_variation' );
+
+	const [ attributes ] = useEntityProp< ProductAttribute[] >(
+		'postType',
+		'product_variation',
+		'attributes'
+	);
+
+	const [ status ] = useEntityProp< string >(
+		'postType',
+		'product_variation',
+		'status'
+	);
+
+	const name = useMemo(
+		() =>
+			attributes
+				.map( ( attribute: ProductAttribute ) => attribute.name )
+				.join( ', ' ),
+		[ attributes ]
+	);
+
+	const { deleteProductVariation } = useDispatch(
+		EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME
+	);
+
+	const { createSuccessNotice, createErrorNotice } =
+		useDispatch( 'core/notices' );
+
+	function handleMenuItemClick() {
+		recordEvent( 'product_dropdown_option_click', {
+			selected_option: 'delete_variation',
+			product_type: 'product_variation',
+			product_id: productId,
+			variation_id: variationId,
+			product_status: status,
+		} );
+
+		setShowModal( true );
+	}
+
+	async function handleRemove() {
+		return deleteProductVariation< Promise< ProductVariation > >( {
```

Comment:
Can we record this event?

Reply:
Do we have the event name and data to record here?

Reply:
In the classic editor we have `wcadmin_product_variations_buttons` with an `action` property named `remove_variation`.
But we could also create a new one here: `wcadmin_product_delete_variation`?

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/40672/commits/e7f3008e6116763987e799a883debc035b3a58ea